### PR TITLE
Add azimuth and dependencies

### PR DESCRIPTION
**Pull request by isuruf**
Describe your pull request here

Adds azimuth package, a Shiny app demonstrating a query-reference mapping algorithm for single-cell data.
----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by isuruf** [comment]
@BiocondaBot please add label

**Comment by daler** [comment]
@isuruf couple of comments.

First, Seurat (`r-seurat`) is [maintained over on conda-forge](https://github.com/conda-forge/r-seurat-feedstock). My best guess would be that's to support Windows users (since bioconda does not build on windows). So any closely-associated packages, like seurat-data and seurat-disk but also likely azimuth should probably also be maintained over there. For example [r-seuratobject](https://github.com/conda-forge/r-seuratobject-feedstock) is closely related and maintained over there.

Second, by convention, R packages take an `r-` prefix to the name. So these should at least be `r-azimuth`, `r-immunogenomics-presto`, `r-seurat-data`, and `r-seurat-disk`.

**Comment by bgruening** [comment]
I hope it's ok to jump in here.

I'm not sure we ever had such a strict rule of cran packages going to `r-` namespace and everything else doesn't, especially because many packages move to cran later when they have been shipped via conda.
What we did instead is (I think) naming all pure R packages `r-`. And this classifies here for your Seurat packages.

I agree that `azimuth` needs to stay here as it has a Bioconductor dependency.
That said, my feeling is that both Seurat packages should move to conda-forge to stay consistent. The others can stay here. But really, as we have no strict rules here, and if you want to maintain them here, I'm also fine with that and we can move them later if the need arises.

**Comment by isuruf** [comment]
I have add the `r-` prefix to all 4 packages. I would prefer to maintain them here as it would be easier to update them together. The two `seurat` pacakges and azimuth are by the same author and the two `seurat` packages are used only in azimuth AFAIK.

File: recipes/r-seurat-disk/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    - r-rlang >=0.4.4
+    - r-stringi >=1.4.6
+    - r-withr >=2.1.2
+test:
+  commands:
+    - '$R -e "library(''SeuratDisk'')"'
+about:
+  home: https://mojaveazure.github.io/seurat-disk/
+  dev_url: https://github.com/mojaveazure/seurat-disk
+  license: GPL-3
```
Comment (bgruening) [inline recipes/r-seurat-disk/meta.yaml:48]: can you please use here SPDX identifiers

**Review by bgruening — APPROVED** [review]
LGTM, if we could use SPDX IDs that would be ace.

**Review by bgruening — APPROVED** [review]

**Comment by isuruf** [comment]
Thanks @bgruening and @daler

**Comment by bgruening** [comment]
Thanks too you for everything you did theast years for the OS community!

**Pull request merged**